src/vtlengine/files/parser/__init__.py

```diff
@@ -143,8 +143,16 @@ def _validate_pandas(
 ) -> pd.DataFrame:
     warnings.filterwarnings("ignore", category=FutureWarning)
     # Identifier checking
+
     id_names = [comp_name for comp_name, comp in components.items() if comp.role == Role.IDENTIFIER]
 
+    missing_columns = [name for name in components.keys() if name not in data.columns.tolist()]
+    if missing_columns:
+        for name in missing_columns:
+            if components[name].nullable is False:
+                raise SemanticError("0-1-1-10", name=dataset_name, comp_name=name)
+        data = data.reindex(columns=list(components.keys()))
```

Comment:
Do not reindex as we may rename already present columns with wrong names.

Suggestion:
```python
data[name] = None